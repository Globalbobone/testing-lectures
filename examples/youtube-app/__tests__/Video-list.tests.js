import videoList from '../Video-list';
import expectedVideos from '../__mocks__/videos.json';

describe('video list', () => {
    it('render empty list', function () {
        expect(videoList({items: [], query: ''})).toMatchSnapshot();
    });
    it('render message for empty result', function () {
        expect(videoList({items: [], query: 'test'})).toMatchSnapshot();
    });
    it('render videos non empty result', function () {
        expect(videoList(expectedVideos)).toMatchSnapshot();
    });
});