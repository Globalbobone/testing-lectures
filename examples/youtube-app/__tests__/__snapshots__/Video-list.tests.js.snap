// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`video list render empty list 1`] = `"<div>Please start search</div>"`;

exports[`video list render message for empty result 1`] = `"<div>Not found</div>"`;

exports[`video list render videos non empty result 1`] = `
"<div>
                <ul>
                <li><img src=\\"thumbnailUrl1\\" alt=\\"\\">Title 1</li><li><img src=\\"thumbnailUrl2\\" alt=\\"\\">Title 2</li>
                </ul>
            </div>"
`;
